### ISSUE TITLE
Nibe + Tibber Pulse

### ISSUE THREAD

**Issue by faanskit** [elupus/hass_nibe#127]
EDIT: This information is **INCORRECT**. See post below. Please close this thread!

Hey,

THIS IS NOT AN ISSUE. 

Don't know where to provide this information, sorry of I use an inappropriate channel. If you know a better place then I'll re-post it there.

With the following configuration, I have created an Individual Energy Meter that is based on the BE1->BE3 sensors from the Nibe integration by combining them with Voltage and Power Factor from the Tibber Pulse.

This makes it possible to integrate the Nibe into the Energy tab of Home Assistant as an individual device. The below configuration obviously need to be changed to match local conditions (names, mapping of phases, etc.)

```
sensor:
  - platform: template
    sensors:
      heatpump_power_watts:
        friendly_name: "Heatpump Power Watts"
        unit_of_measurement: 'W'
        device_class: power
        value_template: >
            {% set PF = states('sensor.power_factor_XXXX') | float %}
            {% set V1 = states('sensor.voltage_phase1_XXXX') | float %}
            {% set V2 = states('sensor.voltage_phase2_XXXX') | float %}
            {% set V3 = states('sensor.voltage_phase3_XXXX') | float %}
            {% set BE1 = states('sensor.nibe_YYYYY_40083') | float %}
            {% set BE2 = states('sensor.nibe_YYYYY_40081') | float %}
            {% set BE3 = states('sensor.nibe_YYYYY_40079') | float %}
            {{ ((PF/100*BE1*V1)+(PF/100*BE2*V2)+(PF/100*BE3*V2)) | round(2) }}

  - platform: integration
    source: sensor.heatpump_power_watts
    name: Energy Spent Heatpump
    unit_prefix: k
    method: trapezoidal
    round: 2

```
/Marcus

**Comment by mist0706** [comment]
This is actually amazing information. Thank you!
By chance do you have a good dashboard that you could share to view the system layout? I'm looking for something similar to this heat exchanger diagram that i've recently started getting to work:
![Screenshot 2022-06-03 at 10 56 02](https://user-images.githubusercontent.com/10143260/171833427-84c0d163-52e9-4a2e-8629-2668fc189cb6.png)

**Comment by faanskit** [comment]
Hi,

This is actually **incorrect** information. I just learned that the _current_ information provided by Nibe is not what the pump consumes, it is input to the pump reads from the sensors that should be installed.

These sensors are used by the pump to select from which phase to take the energy from to avoid blowing the master fuse. In essence, the pump performs load balancing from this.

So, what you get is (should be) the same as Tibber Pulse already provide. In my case, there is ca. 0.5 A difference between the two and the Nibe sensors are only updated once every hour, so I've stopped using these sensors from Nibe.

Sorry for the bad news; I'll update the original post as well.

BR, Marcus
ps. No dashboards. ds.
